src/JointTrajectoryControllerBase.cpp

```diff
@@ -247,7 +250,18 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
       if (goalIt != mGoalConstraints.end())
       {
         std::size_t index = mControlledSkeleton->getIndexOf(dof);
-        if (std::abs(mDesiredPosition[index] - mActualPosition[index]) > (*goalIt).second)
+        auto diff = std::abs(mDesiredPosition[index] - mActualPosition[index]);
+
+        // Check for SO2:
+        auto jointSpace = mControlledSpace->getJointSpace(index);
+        auto r1Joint = std::dynamic_pointer_cast<const R1Joint>(jointSpace);
+        if(!r1Joint) {
+          if(diff > M_PI) {
```

Comment:
Hmm don't we want something that actually does the wrapping and make `diff` be within [-pi, pi]?  Something more like [this? ](https://github.com/personalrobotics/aikido/blob/master/src/statespace/SO2.cpp#L21)